### ISSUE TITLE
MA0011 should not suggest obsolete method

### ISSUE THREAD

**Issue by rsking** [meziantou/Meziantou.Analyzer#544]
### Version of the Meziantou.Analyzer NuGet package

2.0.58

### Rule Identifier

MA0011

### Target Framework

netcoreapp2.0 or greater, netstandard2.0 or greater, net20 or greater

### C# Language version

C#11

### Description

for Enum values, MA0011 suggests that would use an overload with an IFormatProvider. The issue is that the overload with the IFormatProvider is marked as being obsolete.

this analyzer should not suggest an obsolete method. Indeed, probably no analyzer should suggest an obsolete method.

If you take the advice, then this causes a `CS0618` warning, not to use an obsolete method.

### Reproduction Steps

Minimal code:

```c#
_ = System.StringComparison.Ordinal.ToString();
```

**Comment by meziantou** [comment]
I don't reproduce the error. Only MA0052 is reported (use `nameof` instead of `ToString`). Could you provide a repro project? (could be a zip, a github repo, or a CLI step by step)

- `dotnet new classlib`
- `dotnet add package meziantou.analyzer`
- ````
  public class Class1
  {
    void A() => System.StringComparison.Ordinal.ToString();
  }
  ````
- `dotnet build`

=> MA0011 is not reported

**Comment by rsking** [comment]
Hmmm, I do agree that with those steps, I cannot reproduce either. I will try to chase down what specific example I need to do to get this.

![image](https://github.com/meziantou/Meziantou.Analyzer/assets/795030/3ccf26ca-dc8b-4c5b-9ba2-52617f430b1f)

**Comment by rsking** [comment]
OK, I finally found this. It was because I had an extension method to make IFormattable a bit easier, without having to specify a format. i.e.

```c#
public static string? ToString(this IFormattable formattable, IFormatProvider? provider) => formattable.ToString(format: default, provider);
```

The analyzer is correctly picking up the fact that this was using this overload, without the `IFormatProvider` being specified.
